docs/developer-guides/Scripting_Terraform.md

```diff
@@ -0,0 +1,176 @@
+# Developer Guide: Scripting Terraform
```

Comment:
Do we have any guidance here for how to bootstrap the AWS account?
There's a repo [bootstrap-aws](https://github.com/NHSDigital/bootstrap-aws) that will create a s3 bucket and dynamodb state table automatically

Reply:
The [CCoE](https://digital.nhs.uk/services/cloud-centre-of-excellence) will in general do the legwork of getting an account set up for first use, but it might be that we need to do a bit of work on top of what they provide. @stefaniuk do you happen to know if CCoE provide something that's already set up to point terraform at, or do we want to be taking that bit of the config on?  Having tf state Just Work out of the box is quite important

Reply:
Thanks @amaanibn-nasar1-nhs. Do we prefer programmatic creation of the state store over declarative? If we use bootstrap-aws, will the resources it creates be included in the state?

@regularfry Yes, they do the legwork, but setting up the state store is not included in it and has to be done by the team.

Reply:
@amaanibn-nasar1-nhs @regularfry Let's address the question of bootstrapping accounts as a separate topic. There are a number of aspects to consider: 1) Our recommended method of bootstrapping accounts; 2) What should be included in it; 3) Where it all lives and how it's executed; 4) Documentation.